sonic-xcvrd/xcvrd/xcvrd_utilities/port_event_helper.py

```diff
@@ -119,9 +129,19 @@ def handle_port_update_event(self):
                     (port_name, op, fvp) = port_tbl.pop()
                     if not port_name:
                         break
-                    if not validate_port(port_name):
-                        continue
+ 
                     fvp = dict(fvp) if fvp is not None else {}
+                    role = fvp.get(multi_asic.PORT_ROLE, None)
+                    if role:
+                        self.port_role_map[port_name] = role
+                    else:
+                        # role attribute might not be present for state DB
```

Comment:
@vivekrnv at line 85 the map is updated from config_db during __init__

Reply:
Yes, but to handle dpb we need to also check for presence of role attribute here